packages/betterer/src/reader.ts

```diff
@@ -1,20 +1,27 @@
-import { readFile } from 'fs';
-import { promisify } from 'util';
-
+import { existsSync } from 'fs';
 import { BettererResults } from './types';
-
-const readFileAsync = promisify(readFile);
+import * as dedent from 'dedent';
+import to from 'await-to-js';
 
 export async function read(resultsPath: string): Promise<BettererResults> {
-  try {
-    await readFileAsync(resultsPath);
-  } catch {
+  if (!existsSync(resultsPath)) {
```

Comment:
That's _fs.exists_. It's async. What they're warning is that between checking and opening the file could change. That's still true, but to somewhat of a lesser extent because the file checking is synchronous and we get straight into the load afterward. 

This doesn't necessary remove the race condition, but reduces it slightly. 

Also, we're not assuming success after this point. We check to see if it definitively doesn't exist... if that's true, we just return empty. The race condition here would be if it was created in the background while this script was running... I think that's not something we have to worry about too much. 

Afterwards when doing the import we check again. If something happens and it suddenly got deleted (or became inaccessible) between these two steps, we're still doing a check (via the error handling for async/await) of whether we can read it.